### PR TITLE
Statemgmt

### PR DESCRIPTION
**Pull request by Arsnumeralis**
added state management

### CODE REVIEW

**Review by arturasmckwcz — COMMENTED** [review]
Nice

**Review by JauniusZ — APPROVED** [review]

File: www/global.js [hunk 1]
```diff
@@ -1,10 +1,38 @@
...
-  headers,
-  body: JSON.stringify({ query: '{books{title,author}}' })
-}).then(response => response.json());
 
-export const books = result?.data?.books;
+
+export async function setLocalStorage(key, value) {
+  // expiry check to be added
+  const now = new Date();
+  const item = await {
```
[the remark this answers is not in the document]
Comment (Arsnumeralis) [reply www/global.js]: removed in af85a662f036b034c2dbfadd2d6836686844bec6

File: www/global.js [hunk 2]
```diff
@@ -1,10 +1,38 @@
...
+    value: value,
+    // expiry: now.getTime() + expiry
+  }
+  await localStorage.setItem(key, JSON.stringify(item));
+}
+
+let booksObj
```
[the remark this answers is not in the document]
Comment (Arsnumeralis) [reply www/global.js]: essentially yes

File: www/books/books.js [hunk 3]
```diff
@@ -2,7 +2,7 @@ import { books } from '../global.js';
 
 const list = document.getElementById('books');
 
-Array.isArray(books) &&
+await Array.isArray(await books) &&
```
[the remark this answers is not in the document]
Comment (Arsnumeralis) [reply www/books/books.js]: removed await in af85a662f036b034c2dbfadd2d6836686844bec6

File: www/global.js [hunk 4]
```diff
@@ -1,10 +1,38 @@
...
+export async function getBooks(key) {
+  // expiry check to be added
+  const itemStr = await localStorage.getItem(key);
+  if (!itemStr) {
+    const result = await fetch('http://localhost:3000/graphql', {
+      method: 'POST',
+      headers,
+      body: JSON.stringify({ query: '{books{title,author}}' })
+    }).then(response => response.json());
+    booksObj = result.data.books
```
[the remark this answers is not in the document]
Comment (Arsnumeralis) [reply www/global.js]: added try/catch in 408e24f5dde4789869c3297c6f7bd5131db142a3

File: www/global.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (arturasmckwcz) [reply www/global.js]: nit:
I'd consider it as an object named "state" or something similar. And most probably I'd move it to a separate module in a separate file. Also I'd consider methods like set, get, flush, etc.

File: www/global.js [hunk 5]
```diff
@@ -15,14 +15,21 @@ let booksObj;
...
   if (!itemStr) {
-    const result = await fetch("http://localhost:3000/graphql", {
-      method: "POST",
-      headers,
-      body: JSON.stringify({ query: "{books{title,author}}" }),
-    }).then((response) => response.json());
-    booksObj = result?.data?.books;
-    setLocalStorage("books", booksObj, 10000);
-    return booksObj;
+    try {
```
Comment (arturasmckwcz) [inline www/global.js]: this is overhead. you either wrap into try/catch if use async/await or use .then().catch(error). it's an overkill to wrap .then().catch(error) into try/catch

**Review by arturasmckwcz — APPROVED** [review]

**Pull request merged**